crates/sui-framework/packages/bridge/sources/bridge.move

```diff
@@ -217,23 +218,29 @@ module bridge::bridge {
 
     // Record bridge message approvals in Sui, called by the bridge client
     // If already approved, return early instead of aborting.
-    // TODO: rename this to `approve_token_transfer`
-    public fun approve_bridge_message(
+    public fun approve_token_transfer(
         self: &mut Bridge,
         message: BridgeMessage,
         signatures: vector<vector<u8>>,
     ) {
-        // FIXME: need to check pause
         let inner = load_inner_mut(self);
-        let message_key = message.key();
-        // TODO: test this
+        assert!(!inner.paused, EBridgeUnavailable);
+        // verify signatures
+        inner.committee.verify_signatures(message, signatures);
```

Comment:
now i recalled that this was what @patrickkuo  originally wrote but i suggested the current order. The reason was to save gas for clients because there will be multiple clients running for liveness. All of them will get correct signatures but only one of them will trigger the transfer. We wanted to save gas for the remaining clients.

Thoughts?

Reply:
is this a significant amount of gas we are talking about?
between gas and correctness paranoia I vote for charging gas, but maybe we should measure and see what is the true impact.
Thanks for the feedback!

Reply:
also @longbowlu please point me to a test I can run to get a sense of the gas used

Reply:
I think you probably have a better sense than me now :)